From 0d72b693856537290b82c4c19d566fdf4631f04e Mon Sep 17 00:00:00 2001
From: Giovanni Ricca <giovanniricca@proton.me>
Date: Sun, 18 May 2025 17:24:30 +0200
Subject: [PATCH] Revert "sm7250-common: Update partition sizes"

This reverts commit ed6bb854e3ec5b01ab81111b4cb7294ee97c19c3.

sm7250-common: Add system_ext to dynamic partitions list

Change-Id: Ibac5e64c748fdbe2d26acb8fcaa85c674d3ba92f

sm7250-common: Add odm to dynamic partitions list

Change-Id: I173beb983bae67040bc82960a0c053ecb3f9a544
---
 BoardConfigCommon.mk | 10 ++++++----
 common.mk            |  4 +++-
 init/fstab.qcom      |  2 ++
 3 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index c5c8b8cc..ad00861f 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -110,9 +110,7 @@ TARGET_MODULE_ALIASES += \
 TARGET_USES_ION := true
 
 # Partitions
-ifneq ($(WITH_GMS),true)
-BOARD_PRODUCTIMAGE_PARTITION_RESERVED_SIZE := 1610612736
-endif
+BOARD_PRODUCTIMAGE_MINIMAL_PARTITION_RESERVED_SIZE := true
 -include vendor/lineage/config/BoardConfigReservedSize.mk
 
 BOARD_FLASH_BLOCK_SIZE := 262144 # (BOARD_KERNEL_PAGESIZE * 64)
@@ -123,17 +121,21 @@ BOARD_RECOVERYIMAGE_PARTITION_SIZE := 104857600
 
 BOARD_SUPER_PARTITION_SIZE ?= 10736369664
 BOARD_SUPER_PARTITION_GROUPS := moto_dynamic_partitions
-BOARD_MOTO_DYNAMIC_PARTITIONS_PARTITION_LIST := system product vendor
+BOARD_MOTO_DYNAMIC_PARTITIONS_PARTITION_LIST := system system_ext product vendor odm
 BOARD_MOTO_DYNAMIC_PARTITIONS_SIZE ?= 5363990528 # (BOARD_SUPER_PARTITION_SIZE / 2) - 4MB
 
 BOARD_PRODUCTIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_SYSTEMIMAGE_FILE_SYSTEM_TYPE := ext4
+BOARD_SYSTEM_EXTIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_VENDORIMAGE_FILE_SYSTEM_TYPE := ext4
+BOARD_ODMIMAGE_FILE_SYSTEM_TYPE := ext4
 
 TARGET_USERIMAGES_USE_F2FS := true
 
 TARGET_COPY_OUT_PRODUCT := product
+TARGET_COPY_OUT_SYSTEM_EXT := system_ext
 TARGET_COPY_OUT_VENDOR := vendor
+TARGET_COPY_OUT_ODM := odm
 
 # Platform
 BOARD_USES_QCOM_HARDWARE := true
diff --git a/common.mk b/common.mk
index 6cb3739f..f6979ecf 100644
--- a/common.mk
+++ b/common.mk
@@ -19,9 +19,11 @@ AB_OTA_PARTITIONS += \
     product \
     recovery \
     system \
+    system_ext \
     vbmeta \
     vbmeta_system \
-    vendor
+    vendor \
+    odm
 
 PRODUCT_PACKAGES += \
     update_engine \
diff --git a/init/fstab.qcom b/init/fstab.qcom
index 5392a6f5..724be039 100644
--- a/init/fstab.qcom
+++ b/init/fstab.qcom
@@ -9,6 +9,8 @@
 system                                                  /system                  ext4    ro                                                   wait,slotselect,avb=vbmeta_system,avb_keys=/avb/q-gsi.avbpubkey:/avb/r-gsi.avbpubkey:/avb/s-gsi.avbpubkey,logical,first_stage_mount
 vendor                                                  /vendor                  ext4    ro                                                   wait,slotselect,avb=vbmeta,logical,first_stage_mount
 product                                                 /product                 ext4    ro                                                   wait,slotselect,avb,logical,nofail,first_stage_mount
+odm                                                     /odm                     ext4    ro                                                   wait,slotselect,avb=vbmeta,logical,first_stage_mount
+system_ext                                              /system_ext              ext4    ro                                                   wait,slotselect,avb,logical,nofail,first_stage_mount
 /dev/block/by-name/metadata                             /metadata                ext4    noatime,nosuid,nodev,discard,data=journal,commit=1   wait,formattable,first_stage_mount,check,metadata_csum
 /dev/block/bootdevice/by-name/userdata                  /data                    f2fs    noatime,nosuid,nodev,discard,reserve_root=32768,resgid=1065,fsync_mode=nobarrier,inlinecrypt       latemount,wait,check,formattable,fileencryption=aes-256-xts:aes-256-cts:v2+inlinecrypt_optimized,keydirectory=/metadata/vold/metadata_encryption,quota,reservedsize=512M,sysfs_path=/sys/devices/platform/soc/1d84000.ufshc,checkpoint=fs
 /devices/platform/soc/8804000.sdhci/mmc_host*           auto                     auto    defaults                                             wait,voldmanaged=sdcard1:auto
-- 
2.49.0

